src/vuex/modules/index/types.ts

```diff
@@ -1,11 +1,110 @@
+export class Index {
+  private _name: string
+  loading = false
+  collections?: Collection[]
+
+  constructor(name: string) {
+    this._name = name
+  }
+
+  get name() {
+    return this._name
+  }
+
+  get CollectionsCount(): number | undefined {
+    return this.collections?.length
+  }
+
+  public initCollections(collections: Collection[]) {
+    this.collections = collections
+  }
+
+  public addCollection(collection: Collection) {
+    if (!this.collections) {
+      throw new Error(
+        'Unable to perform operations, the collection list is not yet initialized'
+      )
+    }
+    this.collections.push(collection)
+  }
+
+  public removeCollection(collection: Collection) {
```

Comment:
Warning.
You are doing OOP in the store, which is meant to be mutation-oriented. Such an operation will not be tracked by the Vuex DevTools, since it is not performed by a mutation. I really like the OOP-style here, but it is breaking the paradigm, which is the only thing that prevents us from getting crazy on the store (which is a global object, let's not forget it).

Reply:
Maybe you could try to call this method within a mutation, instead of just calling it directly. This way, the benefit of traceability and time-travel provided by Vuex is kept together with the OOP-style separation of concerns.